include/revng/Pipeline/Target.h

```diff
@@ -0,0 +1,359 @@
+  const Kind *K;
+  Exactness Exact;
+
+public:
+  Target(PathComponents Quantifiers,
+         const Kind &K,
+         Exactness Exactness = Exactness::Exact) :
+    Quantifiers(std::move(Quantifiers)), K(&K), Exact(Exactness) {
+    revng_assert(this->Quantifiers.size() == getKind().depth() + 1);
+  }
+
+  Target(Quantifier Quantifier,
+         const Kind &K,
+         Exactness Exactness = Exactness::Exact) :
+    Quantifiers({ std::move(Quantifier) }), K(&K), Exact(Exactness) {
+    revng_assert(this->Quantifiers.size() == getKind().depth() + 1);
+  }
+
+  Target(std::string Name,
+         const Kind &K,
+         Exactness Exactness = Exactness::Exact) :
+    Quantifiers({ Quantifier(std::move(Name)) }), K(&K), Exact(Exactness) {
+
+    revng_assert(this->Quantifiers.size() == getKind().depth() + 1);
+  }
+
+  Target(std::initializer_list<std::string> Names,
+         const Kind &K,
+         Exactness Exactness = Exactness::Exact) :
+    K(&K), Exact(Exactness) {
+    for (auto Name : Names)
+      Quantifiers.emplace_back(std::move(Name));
+    revng_assert(this->Quantifiers.size() == getKind().depth() + 1);
+  }
+
+  Target(llvm::ArrayRef<llvm::StringRef> Names,
+         const Kind &K,
+         Exactness Exactness = Exactness::Exact) :
+    K(&K), Exact(Exactness) {
+    for (auto Name : Names)
+      Quantifiers.emplace_back(Name.str());
+    revng_assert(this->Quantifiers.size() == getKind().depth() + 1);
+  }
+
+  bool operator<(const Target &Other) const {
+    auto Self = std::tie(Quantifiers, K, Exact);
+    auto OtherSelf = std::tie(Other.Quantifiers, Other.K, Other.Exact);
+    return Self < OtherSelf;
+  }
+
+  const Kind &getKind() const { return *K; }
+
+  Exactness kindExactness() const { return Exact; }
+  void setKind(const Kind &NewKind) { K = &NewKind; }
+  void setExactness(Exactness NewExactness) { Exact = NewExactness; }
+
+  const PathComponents &getQuantifiers() const { return Quantifiers; }
+
+  void addGranularity() { Quantifiers.emplace_back(); }
+  void dropGranularity() { Quantifiers.pop_back(); }
+
+  template<typename OStream>
+  void dump(OStream &OS, size_t Indents = 0) const debug_function {
+    indent(OS, Indents);
+
+    OS << (Exact == Exactness::DerivedFrom ? "derived from " : "exactly ")
+       << K->name().str() << " With path: ";
+    for (const auto &Entry : Quantifiers) {
+      Entry.dump(OS);
+      OS << "/";
+    }
+    OS << "\n";
+  }
+
+  template<typename OStream, typename Range>
+  static void dumpQuantifiers(OStream &OS, Range R) debug_function {
+    for (const auto &Entry : R) {
+      Entry.dump(OS);
+      OS << "/";
+    }
+  }
+
+  void expand(const Context &Ctx, TargetsList &Out) const {
+    K->expandTarget(Ctx, *this, Out);
+  }
+
+  std::string serialize() const;
+
+  void dump() const debug_function { dump(dbg); }
+
+  int operator<=>(const Target &Other) const;
+
+  bool operator==(const Target &Other) const { return (*this <=> Other) == 0; }
```

Comment:
Why do we need this if there's `<=>`.

Reply:
i don't know, it did not generated it automatically, i understand it's something about that it can generate them only if a std::ordering_something is returned by <=> but those are not in the standard library yet